src/vm/memory.zig

```diff
@@ -85,17 +129,11 @@ pub const Region = struct {
 
     /// Get the underlying host slice of memory.
     ///
-    /// Returns an error if you're trying to get mutable access to a constant region.
-    pub fn getSlice(self: Region, comptime state: MemoryState) !state.Slice() {
-        return switch (state) {
-            .constant => switch (self.host_memory) {
-                .constant => |constant| constant,
-                .mutable => |mutable| mutable,
-            },
-            .mutable => switch (self.host_memory) {
-                .constant => return error.AccessViolation,
-                .mutable => |mutable| mutable,
-            },
+    /// Returns null if you're trying to get mutable access to a constant region.
+    pub fn hostSlice(self: Region, comptime state: MemoryState) ?state.Slice() {
+        return switch (self.host_memory) {
+            .constant => |constant| if (state == .mutable) null else constant,
```

Comment:
feels like this null case should be an error -- wdyt

Reply:
We need additional context (version + config, which Region code is not aware of) to return the correct error value, so it's best to resolve the error elsewhere.